test/unit-test/mqtt_agent_utest.c

```diff
@@ -1267,20 +1308,40 @@ void test_MQTTAgent_CommandLoop_add_acknowledgment_failure( void )
     commandToSend.pCmdContext = NULL;
     commandToSend.pArgs = NULL;
 
+    globalMessageContext.pSentCommand = &commandToSend;
+
+    /* Test case when the list of pending acknowledgements is full. */
     for( i = 0; i < MQTT_AGENT_MAX_OUTSTANDING_ACKS; i++ )
     {
         /* Assigning valid packet ID to all array spaces to make no space for incoming acknowledgment. */
-        mqttAgentContext.pPendingAcks[ i ].packetId = i + 1;
+        mqttAgentContext.pPendingAcks[ i ].packetId = testPacketId + i + 1;
     }
 
-    globalMessageContext.pSentCommand = &commandToSend;
-
+    /* Call API under test. */
     mqttStatus = MQTTAgent_CommandLoop( &mqttAgentContext );
 
     TEST_ASSERT_EQUAL( MQTTNoMemory, mqttStatus );
 
     /* Ensure that callback is invoked. */
     TEST_ASSERT_EQUAL( 1, commandCompleteCallbackCount );
+
+    MQTTAgentCommand_Publish_ExpectAnyArgsAndReturn( MQTTSuccess );
```

Comment:
Can you reset the value of `commandCompleteCallbackCount` right here? It looks like the only reason it should be 2 later is that a test was run once before.

Reply:
Sure, done.